TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1129,3 +1129,33 @@ def exterior_boundaries(feature_layers, zoom,
         new_layer['name'] = new_layer_name
 
         return new_layer
+
+
+def generate_label_features(feature_layers, zoom, source_layer=None,
+                            label_property_name=None,
+                            label_property_value=None):
+    assert source_layer, 'generate_label_features: missing source_layer'
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    new_features = []
+    for feature in layer['features']:
+        shape, properties, fid = feature
+
+        # shapely does the right thing for all kinds of geometries
+        # it also has a function `representative_point` which we might
+        # want to consider using too
+        label_centroid = shape.centroid
```

Comment:
mapshaper is better at this, btw. We've been using it for Who's On First and it more reliably results in a lat, lng pair in the "middle" of a feature. But that introduces a new dependency, and is an optimization – so let's add a ticket to the backlog for this.